scripts/docs_config/glossary.rst

```diff
@@ -0,0 +1,97 @@
+Glossary
+==========================================================
+
+Homogeneous Memory
+  A collection of memory composed of a single memory type, managed by a singular 
+  driver using a uniform approach.
+
+Heterogeneous Memory
+  A set of memory composed of multiple types of memory technologies, each 
+  requiring distinct handling approaches often managed by separate drivers.
+
+Memory Tiering
+  An organization and hierarchy of different types of memory storage within a 
+  system, with each type of memory having distinct characteristics, performance, 
+  and cost attributes. These memory tiers are typically organized in a 
+  hierarchy, with faster, more expensive memory located closer to the processor 
+  and slower, less expensive memory located further away.
+
+Memory Access Initiator 
+  A component in a computer system that initiates or requests access to the 
+  computer's memory subsystem. This could be a CPU, GPU, or other I/O and cache 
+  devices.
+
+Memory Target 
+  Any part of the memory subsystem that can handle memory access requests. This 
+  could be the OS memory (RAM), video memory that resides on the graphics 
+  cards, memory caches, storage, external memory devices connected using 
+  CXL.mem protocol, etc.
+
+Memory Page 
+  A fixed-length contiguous block of virtual memory, described by a single 
+  entry in the page table. It is the smallest unit of data for memory 
+  management in a virtual memory operating system.
+
+Enlightened Application 
+  An application that explicitly manages data allocation distribution among 
+  memory tiers and further data migration. 
+
+Unenlightened Application 
+  An application that coexists with the underlying infrastructure (OS, 
+  frameworks, libraries) that offers various memory tiering and migration 
+  solutions without any code modifications.
+
+Memory Pool 
+  A memory management technique used in computer programming and software 
+  development, where fixed-size blocks of memory are preallocated using one or 
+  more memory providers and then divided into smaller, fixed-size blocks or 
```

Comment:
I think they don't have to be divided into fixed-size blocked. For example, consider something like a bump allocator

Reply:
I removed "fixed-size"